### PR TITLE
SRS

### LINKED ISSUE #1: Prepare SRS
Prepare a document with all requirements

### PR DESCRIPTION
**Pull request by MykytaKyt**
Add SOFTWARE REQUIREMENTS SPECIFICATION
Fixes https://github.com/NikitaKit1998/datakyt/issues/1

### CODE REVIEW

File: docs/srs.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
+# SOFTWARE REQUIREMENTS SPECIFICATION
+
+## Prepared by
+
+### Mykyta Kyt | Alex Zhytnyk
+
+1. Introduction
```
Comment (aartiukh) [inline docs/srs.md]: Use links in the content or just remove it

File: docs/srs.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+An accounting database system stores the following information.
+
+- **Equipment details:** It includes information about the equipment, such as laptops, monitors, mice, keyboards, headphones, which are issued to employees, as well as information about the equipment that is needed in other cities.
+- **Employee description:** It includes the employee's id, first name, last name, position, mail, phone number, birthday, city, project.
+- **Description of interaction:** It includes information which technique is given to each employee or information that the employee works on his technique. It also includes information about the workplace.
+
+### 2.2 Product features
+
+This item presents the DFD-diagram, which shows the general cooperation in the system. They are presented in pictures 1.1-1.2 [![Picture 1.1 - DFD diagram A-0 level.](https://i.ibb.co/Jjn6rqM/Untitled-Document.png)](https://ibb.co/r2k8531)
```
Comment (aartiukh) [inline docs/srs.md]: Store images in the repo

File: docs/srs.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+Picture 1.1 - DFD diagram A-0 level.
+
+[![Picture 1.2 - DFD diagram A0 level.](https://i.ibb.co/KDq4P58/DFD-A-0-1.png)](https://ibb.co/nBcdGwK)
```
Comment (aartiukh) [inline docs/srs.md]: Store images in the repo

File: docs/srs.md [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+5. <https://hydra.cc/docs/intro/> : Hydra is an open-source Python framework that simplifies the development of research and other complex applications.
+
+## 2\. Overall Description
+
+### 2.1 Product perspective
+
+An accounting database system stores the following information.
+
+- **Equipment details:** It includes information about the equipment, such as laptops, monitors, mice, keyboards, headphones, which are issued to employees, as well as information about the equipment that is needed in other cities.
+- **Employee description:** It includes the employee's id, first name, last name, position, mail, phone number, birthday, city, project.
```
Comment (aartiukh) [inline docs/srs.md]: Why do we need a birthday?

File: docs/srs.md [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+- **Description of interaction:** It includes information which technique is given to each employee or information that the employee works on his technique. It also includes information about the workplace.
```
Comment (aartiukh) [inline docs/srs.md:55]: what about date when employee receives equipment? think about other missed fields.

File: docs/srs.md [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+### 2.3 Operating environment
+
+This product will work in any browser, as it will represent a web page. On PostgreSQL 9.5, Python 3.8\. This database will be located on Amazon AWS servers.
+
+### 2.4 Design and implementation constraints
+
+Our constraints are:
+
+- Possibilities of Python programming language and PostgreSQL DBMS;
+- Writing Python code that conforms to PEP 8;
```
Comment (aartiukh) [inline docs/srs.md]:
```suggestion
- Writing Python code that satisfies PEP 8;
```

**Review by aartiukh — CHANGES_REQUESTED** [review]

File: docs/srs.md (hunk 1 above)
Comment (MykytaKyt) [reply docs/srs.md]: Okay

File: docs/srs.md (hunk 4 above)
Comment (MykytaKyt) [reply docs/srs.md]: Because, in the employee table that I have, there is such a field.

File: docs/srs.md (hunk 2 above)
Comment (MykytaKyt) [reply docs/srs.md]: How to add it there?

File: docs/srs.md (hunk 5 above)
Comment (MykytaKyt) [reply docs/srs.md:55]: Okay

File: docs/srs.md (hunk 6 above)
Comment (MykytaKyt) [reply docs/srs.md]: Corrected

File: docs/srs.md (hunk 4 above)
Comment (aartiukh) [reply docs/srs.md]: This is not a valid reason to store unneeded info. Also let's keep personal info as less as possible

File: docs/srs.md (hunk 2 above)
Comment (aartiukh) [reply docs/srs.md]: Create a folder `images`, put all needed images there and commit.
In MD use relative path
```
![Image Name](images/image.png)
```

**Review by aartiukh — CHANGES_REQUESTED** [review]
https://github.com/NikitaKit1998/datakyt/pull/2/files#r502808505

**Review by aartiukh — DISMISSED** [review]

**Review by OleksandrZhytnyk — DISMISSED** [review]

File: docs/srs.md [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+
+- **Import/Export** to .csv file/ .xls file: saving or uploading tables.
+
+- **Generating QR codes** with unique inventory number: unique number on the inventory, which will make it easier to count and keep records
+
+- **Back up**: Saving the database before entering and changing information.
+
+- **Data Structuring**: all information in a digital repository is organized into a clear hierarchical structure with records, tables or objects.
+- **Multi-user Access**: provides multiple users access to all types of information stored in one and the same data store.
+- **Add/Remove** information about employee's or equipment in the database.
```
Comment (paulzcooper) [inline docs/srs.md]:
```suggestion
- **Add/Remove** information about employees or equipment in the database.
```

File: docs/srs.md [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+- **User-Friendly**: read-only access to the database.
```
Comment (paulzcooper) [inline docs/srs.md]: Do you consider 'user-friendly' as a feature or as a quality?  Is it actually should appear in the Feature description section?

File: docs/srs.md [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+- **Reporting tools**: necessary data query
+- **Inventory Monitoring**: the activity of checking stock.
+
+## 4\. External interface requirements
+
+In my opinion, the application will intercommunicate directly through the web page. That is, to enter the site, login depending on the level of access (ordinary user, administrator), and continue to perform various operations, such as adding/removing employees or technicians, creating reports on the number of technicians or employees.
```
Comment (paulzcooper) [inline docs/srs.md]: > the application will intercommunicate directly through the web page

With whom the application will intercommunicate directly? Did you mean a user or the database server?

> That is, to enter the site, login depending on the level of access (ordinary user, administrator)

Does it mean that every user will have to log in to be able to use the application or authorization will only be required for the admin role?

File: docs/srs.md [hunk 10]
```diff
@@ -0,0 +1,130 @@
...
+
+### 4.1 User interfaces
+
+- Front - end software: Python 3.8 version
+- Back - end software: PostgreSQL 9.5 version
+
+### 4.2 Hardware interfaces
+
+- a browser
+
```
Comment (paulzcooper) [inline docs/srs.md:115]: Could you please clarify why there are Python & Postgres in the 'user interfaces section' and browser in the hardware section as soon as it's python & Postgres which are not visible for user and browser which is visible? Shall it be vice versa? I'm not familiar with SRS terminology, but a browser in the hardware section looks weird to me, so I'd rather named it the 'software interfaces' or something (if it remains to be there)

File: docs/srs.md [hunk 11]
```diff
@@ -0,0 +1,130 @@
...
+## 5\. Non functional requirements
+
+### 5.1 Software quality attributes
+
+For any system that is implemented in the software, several checks must be performed to minimize subsequent errors in its use. In our case we have used the following checks to detect errors:
```
Comment (paulzcooper) [inline docs/srs.md]:
```suggestion
For any system that is implemented in the software, several checks must be performed to minimize subsequent errors in its usage. In our case we have used the following checks to detect errors:
```
Am I right, our guiding b2 expert? @aartiukh you taught me that word

**Review by paulzcooper — CHANGES_REQUESTED** [review]

File: docs/srs.md (hunk 7 above)
Comment (OleksandrZhytnyk) [reply docs/srs.md]: Ok

File: docs/srs.md (hunk 8 above)
Comment (OleksandrZhytnyk) [reply docs/srs.md]: Ok, we will move it to other section

File: docs/srs.md (hunk 9 above)
Comment (OleksandrZhytnyk) [reply docs/srs.md]: 1) Users will communicate with the database through website.
2)Two types of users: read-only or full access.

File: docs/srs.md (hunk 10 above)
Comment (OleksandrZhytnyk) [reply docs/srs.md:115]: Our Head Master said that it is correct.

**Review by OleksandrZhytnyk — DISMISSED** [review]

File: docs/srs.md (hunk 10 above)
Comment (MykytaKyt) [reply docs/srs.md:115]: image

File: docs/srs.md (hunk 9 above)
Comment (MykytaKyt) [reply docs/srs.md]: Fixed

File: docs/srs.md [hunk 12]
```diff
@@ -0,0 +1,131 @@
...
+- **Multi-user Access**: provides multiple users access to all types of information stored in one and the same data store.
+
+- **Add/Remove** information about employees or equipment in the database.
+
+- **Reporting tools**: necessary data query
+- **Inventory Monitoring**: the activity of checking stock.
+
+## 4\. External interface requirements
+
+In my opinion, the application will intercommunicate directly through the web page with users. That is, to enter the site, login depending on the level of access (ordinary user, administrator this means that there will be an account for the admin and the ordinary user), and continue to perform various operations, such as adding/removing employees or technicians, creating reports on the number of technicians or employees.
```
Comment (paulzcooper) [inline docs/srs.md]:
```suggestion
In my opinion, the application will intercommunicate directly through the web page with users. That is, to enter the site, login is required. Depending on the level of access (ordinary user or administrator -- this means that there will be an account for the admin and the ordinary user), the user continues to perform various operations, such as adding/removing employees or technicians, creating reports on the number of technicians or employees.
```

**Review by paulzcooper — APPROVED** [review]

**Pull request merged**